src/model.jl

```diff
@@ -91,23 +91,29 @@ function _make_bus_demand_flexibility_amount(
 )::Tuple{Matrix,Matrix}
     # Bus weighting
     zone_to_bus_shares = _make_bus_demand_weighting(case)
-
     # Demand flexibility up profile
     simulation_demand_flex_amt_up = Matrix(
         demand_flexibility.flex_amt_up[start_index:end_index, 2:end]
     )
-    bus_demand_flex_amt_up = permutedims(
-        simulation_demand_flex_amt_up * zone_to_bus_shares
-    )
-
     # Demand flexibility down profile
     simulation_demand_flex_amt_dn = Matrix(
         demand_flexibility.flex_amt_dn[start_index:end_index, 2:end]
     )
-    bus_demand_flex_amt_dn = permutedims(
-        simulation_demand_flex_amt_dn * zone_to_bus_shares
-    )
-
+    # convert per-area input to per-bus values
+    if demand_flexibility.cost_input_type == "AREA"
+        bus_demand_flex_amt_up = permutedims(
+            simulation_demand_flex_amt_up * zone_to_bus_shares
+        )
+        bus_demand_flex_amt_dn = permutedims(
+            simulation_demand_flex_amt_dn * zone_to_bus_shares
+        )
+    # input is already per-bus, copy to new container
+    elseif demand_flexibility.cost_input_type == "BUS"
+        bus_demand_flex_amt_up = transpose(simulation_demand_flex_amt_up)
+        bus_demand_flex_amt_dn = transpose(simulation_demand_flex_amt_dn)
```

Comment:
I'm not fully clear on the merits of using `permutedims` over `transpose`, but we seem to almost exclusively use `permutedims`. I know that `permutedims` and `transpose` yield the same result but return different data types: I believe that `permutedims` returns a value that is of type `Array` while `transpose` returns a value that is of type `LinearAlgebra` (or something like that). This shouldn't be a problem because we are not inputting `bus_demand_flex_amt_up` or `bus_demand_flex_amt_dn` into any other functions, so I don't believe there should be a problem with conflicting input types. Is there a reason we generally use `permutedims`?

Reply:
I think `transpose` returns a view into the array, while `permutedims` returns a new object, so for safety we use `permutedims` so that if we modify the result, we don't modify the upstream object.

Reply:
sure I will change them to permutedims